substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/JavaMainWrapper.java

```diff
@@ -133,6 +135,12 @@ public static int runCore(int paramArgc, CCharPointerPointer paramArgv) {
         int exitCode;
         try {
             if (SubstrateOptions.ParseRuntimeOptions.getValue()) {
+                if (SubstrateUtil.isPosix()) {
+                    if (SubstrateOptions.InstallExitHandlers.getValue()) {
```

Comment:
but the check for || (SubstrateUtil.isLinux() && ProcessProperties.getProcessId() ==1) is missing here? Otherwise how do we get out-of-the-box correct handling for the Docker init process?

Reply:
I decided against making the installation of the handlers dependent on  `ProcessProperties.getProcessId() == 1`.

I tested what the JVM does when running in docker and without. Regarding signal handling the JVM makes no distinction whatsoever. So to be as close as possible to the behaviour of the JVM I do not predicate on PID1. Also since we require the user to opt-in via `--install-exit-handlers` there should be no regressions for anyone not interested in having this JVM-like behaviour for SIGHUP/SIGINT/SIGTERM exit handling.

Reply:
OK. Having a different automatic behavior when running as Docker pid 1 is probably a bad idea.